packages/manager/apps/web/client/app/hosting/cdn/order/hosting-cdn-order.controller.js

```diff
@@ -1,56 +1,109 @@
+import get from "lodash/get";
 
 export default class {
   /* @ngInject */
-  constructor($filter, $timeout) {
+  constructor($scope, $filter, $timeout, $translate, HostingCdnOrderService) {
+    this.$scope = $scope;
     this.$filter = $filter;
     this.$timeout = $timeout;
+    this.$translate = $translate;
+    this.HostingCdnOrderService = HostingCdnOrderService;
+    this.cdnCase = null;
+    this.cdn1Price = null;
   }
 
   $onInit() {
     // Auto-select duration
+    console.log('ZM:: serviceInfo', this.serviceInfo);
+    console.log('ZM:: catalogAddon', this.catalogAddon);
+    console.log('ZM:: catalogAddons', this.catalogAddons);
     this.prices = filter(this.catalogAddon.pricings, ({ capacities }) =>
       includes(capacities, 'renew'),
     );
     [this.price] = this.prices;
     this.interval = this.price.interval;
     this.isEditable = true;
+    console.log('ZM:: price', this.price);
 
     if (this.catalogAddon.pricings.length === 1) {
       // Go directly to the next step
       this.currentIndex = 1;
       this.isEditable = false;
     }
+
+    if (this.isIncludedCDN) {
+      this.cdnCase = 'included';
+    } else if (this.isPayableCDN) {
+      this.cdnCase = 'payable';
+    } else if (!this.hasCDN) {
+      this.cdnCase = 'without';
+    }
+
+    const cdn1Addon = this.catalogAddons.addons.find(addon => addon.planCode === "cdn_business");
+    const cdn1Prices = filter(cdn1Addon.pricings, ({ capacities }) =>
+      includes(capacities, 'renew'),
+    );
+    [this.cdn1Price] = cdn1Prices;
+
+    console.log('ZM:: cdn1Price', cdn1Addon, this.cdn1Price);
+    console.log('ZM:: cdnCase', this.cdnCase);
   }
 
   resetCart() {
     if (this.cart) {
       this.cart = undefined;
       this.cartId = undefined;
+      this.addonPlan = undefined;
+      this.serviceId = undefined;
     }
   }
 
   async prepareCheckout() {
+    console.log('ZM:: prepareCheckout start');
+
     if (!this.cart && !this.checkoutLoading) {
       this.checkoutLoading = true;
-      const { cart, cartId } = await this.prepareOrderCart();
+      const prepareToOrderCart = !this.hasCDN ? this.prepareOrderCart : this.prepareUpgradeOrderCart;
+      const { cart, cartId, addonPlan, serviceId } = await prepareToOrderCart();
+
+      console.log('ZM:: prepareCheckout', cartId, cart);
 
       this.$timeout(() => {
         this.cart = cart;
         this.cartId = cartId;
+        this.addonPlan = addonPlan;
+        this.serviceId = serviceId;
         this.checkoutLoading = false;
       });
+      console.log('ZM:: prepareCheckout end');
     }
   }
 
   checkout() {
+    console.log('ZM:: checkout step');
     this.checkoutLoading = true;
 
-    this.checkoutOrderCart(!!this.defaultPaymentMethod, this.cartId);
+    if (!this.hasCDN) {
+      this.checkoutOrderCart(!!this.defaultPaymentMethod, this.cartId);
+    } else {
+      this.checkoutOrderCartForUpgrade(!!this.defaultPaymentMethod, this.addonPlan, this.serviceId);
+    }
```

Comment:
I'm wondering if we should have two components instead and depending on the use case we display either one or the other as it seems there are a lot of `if/else` logic

Reply:
fixed